### PR TITLE
BAAS-24002: Expose public lazy load function

### PR DESCRIPTION
**Pull request by andrewzlchen**

### CODE REVIEW

**Review by arahmanan — APPROVED** [review]

**Review by Calvinnix — APPROVED** [review]
🚀

File: native.go [hunk 1]
```diff
@@ -195,6 +195,20 @@ func (n NativeClass) InstanceOf(val interface{}) Value {
...
+	}
+
+	createObjectImpl := func(val *Object) objectImpl {
+		obj := create(val)
+		return obj.self
+	}
+
+	return r.newLazyObject(createObjectImpl), nil
+}
+
```
Comment (Gabri3l) [inline native.go:211]: While I know this is a trivial function we have gotten into the bad habit of not adding unit tests in goja. I'd like to change that, do you mind adding a simple table test for this function?

**Review by Gabri3l — CHANGES_REQUESTED** [review]
Would like to see some testing around this, can't wait to play with it!

File: native.go (hunk 1 above)
Comment (andrewzlchen) [reply native.go:211]: hmm yeah, that's a good point! I can definitely do this 👍

File: native.go [hunk 2]
```diff
@@ -195,6 +195,20 @@ func (n NativeClass) InstanceOf(val interface{}) Value {
 	return obj
 }
 
+// NewLazyObject creates a lazy object that will do initialization when the object is called upon during runtime
+func (r *Runtime) NewLazyObject(create func(val *Object) *Object) (Value, error) {
+	if create == nil {
+		return UndefinedValue(), errors.New("create cannot be nil")
+	}
```
Comment (Gabri3l) [inline native.go:202]: [minor] Missing this case in the tests

**Review by Gabri3l — APPROVED** [review]
I really like the test cases you added! Wanted to cover the `nil` case and we're on the clear 👍🏼

**Pull request merged**